git.md

```diff
@@ -153,7 +153,184 @@ Some commit title ISSUE-51 #resolve #time 2h30m
   public always ensure that it conforms to the style-guide. If it doesn't:
   squash, rebase, reword commits as necessary.
 
-### 4. Misc.
+### 4. Rewriting history
+
+#### 4.1 Rewording last commit
+
+Perhaps the most straightforward history rewrite is changing the commit
+message. If the commit in question is the *last* one, the task is trivial:
+
+```git
+git commit --amend
+```
+
+This command will open an editor (configured with `git config core.editor`)
+with last commit for you to modify. After the changes are saved, git will
+update your *local repository*.
+
+---
+Please note that this will *only* work if you didn't push your changes to the
+remote. Before pushing your changes, always ensure that changes are ready.
```

Comment:
What do you mean by "changes are ready"? I'd rephrase and reorder those sentences to something like:

> Please note that this will only work if you didn't push your changes to the remote. In other cases, you will have to use `push --force` which is invasive and dangerous operation. 

Before pushing your changes do your best to make sure they are ready to be published to avoid the necessity to rewrite the history.